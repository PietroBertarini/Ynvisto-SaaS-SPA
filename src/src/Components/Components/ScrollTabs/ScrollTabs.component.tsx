import React from 'react';
import AppBar from '@material-ui/core/AppBar';

import { Link } from 'react-router-dom';
import { StyledTab, StyledTabs, useStyles } from './ScrollTabs.styles';

// https://blog.logrocket.com/routes-animation-transitions-in-react-router-v4-9f4788deb964/ for animation
export interface ScrollTabsProps {
    /**
     * Initial Selected Tab value(First = 0, Second = 1...)
     */
    initialTabValue: number;
    /**
     * All tabs labels.
     */
    tabsLabel: string[];
}

/**
 * Scroll tabs for change page inside Ynvisto.
 */
function ScrollTabs(props: ScrollTabsProps) {
  const { initialTabValue, tabsLabel } = props;
  const classes = useStyles();
  const [value, setValue] = React.useState(initialTabValue);

  const handleChange = (event: React.ChangeEvent<{}>, newValue: number) => {
    setValue(newValue);
  };

  return (
    <div className={classes.root}>
      <AppBar position="static" color="transparent" elevation={0}>
        <StyledTabs
          value={value}
          onChange={handleChange}
          className={classes.styledTabs}
          indicatorColor="secondary"
          textColor="secondary"
          variant="scrollable"
          scrollButtons="auto"
          aria-label="scrollable auto tabs example"
        >
          {tabsLabel.map((tabLabel :string, index) => (
            <StyledTab
              to={`${tabLabel.replace('/', '%20')}`}
              component={Link}
              label={tabLabel}
              key={tabLabel}
              {...a11yProps(index)}
            />
          ))}
        </StyledTabs>
      </AppBar>
    </div>
  );
}
function a11yProps(index: any) {
  return {
    id: `scrollable-auto-tab-${index}`,
    'aria-controls': `scrollable-auto-tabpanel-${index}`,
  };
}

export default ScrollTabs;
