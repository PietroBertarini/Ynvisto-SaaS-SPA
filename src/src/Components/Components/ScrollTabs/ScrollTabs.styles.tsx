import {
  makeStyles, withStyles, Theme,
} from '@material-ui/core/styles';
import Tab from '@material-ui/core/Tab';
import React from 'react';
import Tabs from '@material-ui/core/Tabs';

export const useStyles = makeStyles((theme: Theme) => ({
  root: {
    flexGrow: 1,
    width: '100%',
  },
  styledTabs: {
    '& .MuiTabs-scrollButtonsDesktop': {
      color: '#B1B1B1',
      '&:hover': { color: '#FFFFFF' },
    },
    '& .MuiTabs-flexContainer': {
      // justifyContent: 'center', //save for NotScrollBar
    },
  },
}));

interface AntTabProps {
  label: string;
  to:string;
  component: any;

}
export const StyledTab = withStyles((theme) => ({
  root: {
    textTransform: 'none',
    minWidth: 72,
    maxWidth: '50vw',
    fontSize: '1.6vw',
    color: '#B1B1B1',
    backgroundColor: 'transparent',
    fontWeight: theme.typography.fontWeightRegular,
    marginRight: theme.spacing(4),
    '&:hover': {
      color: '#FFFFFF',
      opacity: 1,
    },
    '&$selected': {
      color: '#FFFFFF',
      fontWeight: theme.typography.fontWeightMedium,
    },
    '&:focus': {
      color: '#FFFFFF',
    },
  },
  selected: {},
}))((props: AntTabProps) => <Tab disableRipple {...props} />);

interface StyledProps {
  value: number;
  onChange: any;
  indicatorColor:string;
  textColor:string;
  className: any;
  variant:string;
  scrollButtons:string;
}

export const StyledTabs = withStyles({
  indicator: {
    display: 'flex',
    justifyContent: 'center',
    backgroundColor: 'transparent',
    '& > span': {
      width: '100%',
      backgroundColor: '#BDA058',
    },
  },
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
// @ts-ignore
})((props: StyledProps) => <Tabs {...props} TabIndicatorProps={{ children: <span /> }} />);
