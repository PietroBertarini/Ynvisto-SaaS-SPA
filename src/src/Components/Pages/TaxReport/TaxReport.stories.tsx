import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import ProfitAndDarf from './TaxReport.component';

export default {
  title: 'Services/TaxReports',
  component: ProfitAndDarf,
} as Meta;

const Template: Story<{}> = (args) => (<ProfitAndDarf {...args} />
);

export const Default = Template.bind({});
