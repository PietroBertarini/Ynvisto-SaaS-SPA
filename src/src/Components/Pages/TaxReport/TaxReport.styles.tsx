import { createStyles, makeStyles } from '@material-ui/core/styles';
import theme from '../../../Theme/theme';

export const useStyles = makeStyles(() => createStyles({
  toolbar: theme.mixins.toolbar,
  root: {
    backgroundColor: 'linear-gradient(35.82deg, #000000 -28.37%, #0F0F0F 81.41%, #32323E 93.95%)',
    flexGrow: 1,
    marginLeft: 239,
  },
  header: {
    color: '#FFFFFF',
    fontSize: '2.8vw',
    fontFamily: 'Zurich',
  },
  subTitle: {
    color: '#FFFFFF',
    fontSize: '1.9vw',
    marginBottom: '1vw',
    marginTop: '1vw',
  },
}));
