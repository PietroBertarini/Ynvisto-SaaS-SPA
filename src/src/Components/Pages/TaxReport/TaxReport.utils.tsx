import { EProfitCardStatus, ProfitCardProps } from '../../Components/ProfitCard/ProfitCard.component';
import { EResumeStatus, ResumoInfoProps } from '../../Components/ResumeInfo/ResumeInfo.component';
import MonthlyTaxReport from '../../../Classes/MonthlyTaxReport/MonthlyTaxReport.class';

const otherTabs = [
  'Geral',
];

export const allMonths = [
  'Janeiro',
  'Fevereiro',
  'Março',
  'Abril',
  'Maio',
  'Junho',
  'Julho',
  'Agosto',
  'Setembro',
  'Outubro',
  'Novembro',
  'Dezembro',
];
// TODO: Definitely get from api in future
const DEFAULT_STOCK_THRESHOLD = 20_000;

/** All navigation tabs inside TaxReport screen  */
export const allTabs = otherTabs.concat(allMonths);

/**
 * Get all profit data and tax data to populate the monthly calendar of TaxAndReport page.
 * @param monthlyReports: Monthly report data from api.
 */
export function buildProfitCardsProps(monthlyReports ?: MonthlyTaxReport[])
    : ProfitCardProps[] {
  if (monthlyReports === undefined) {
    return createRemainingProfitCardsProps([]);
  }
  const actualYear = new Date().getFullYear();
  const profitAndTaxFromApi = monthlyReports.filter((item) => item.year
      === actualYear).map(buildProfitCardProps);
  return createRemainingProfitCardsProps(profitAndTaxFromApi);
}

/**
 * Get a individual profit data and tax data to populate the monthly calendar of TaxAndReport page.
 * @param monthlyReport: Monthly report data from api.
 */
export const buildProfitCardProps = (monthlyReport : MonthlyTaxReport) : ProfitCardProps => {
  const actualMonth = new Date().getMonth();
  const monthlyReportMonth = monthlyReport.monthIndex;

  const { tax } = monthlyReport;
  const { profit } = monthlyReport;

  return {
    profit: monthlyReportMonth > actualMonth ? 0 : profit,
    month: allMonths[monthlyReportMonth],
    tax: monthlyReportMonth > actualMonth ? 0 : tax,
    status: getTaxStatus(tax, monthlyReportMonth, !!monthlyReport.darfPaymentId),
  };
};

export function getDefaultProfitCard(month: number) : ProfitCardProps {
  const actualMonth = new Date().getMonth();
  return {
    profit: 0,
    month: allMonths[month],
    tax: 0,
    status: month >= actualMonth ? EProfitCardStatus.NOT_FINISHED : EProfitCardStatus.NO_DARF,
  };
}

export function createRemainingProfitCardsProps(builtProps : ProfitCardProps[]) : ProfitCardProps[] {
  // TODO: This executes in O(n^2). Pass this to O(n)
  return allMonths.map((month, index) => {
    const propsFound = builtProps.find((item) => item.month === allMonths[index]);
    return (
      propsFound === undefined ? getDefaultProfitCard(index) : propsFound);
  });
}

/**
 * Create the Resume Info of Tax Report page using the actual month tax and profit data.
 * @param monthlyReport: Actual monthly data.
 */
export function getResumeInfo(monthlyReport: MonthlyTaxReport[] | undefined) : ResumoInfoProps[] {
  const actualMonth = new Date().getMonth();
  const defaultResume = [{
    value: DEFAULT_STOCK_THRESHOLD,
    subtitle: `Volume isento disponível em ${allMonths[actualMonth]}`,
    status: EResumeStatus.GOOD,
  }];
  if (!monthlyReport) {
    return defaultResume;
  }
  const monthlyReportActualMonth = monthlyReport.find(
    (item) => item.monthIndex === actualMonth,
  );

  if (!monthlyReportActualMonth) {
    return defaultResume;
  }

  const remainVolume = monthlyReportActualMonth.remainExemptionVolumeBrazil;
  return [{
    value: remainVolume < 0 ? 0 : remainVolume,
    subtitle: `Volume isento disponível em ${allMonths[actualMonth]}`,
    status: getStatusResume(monthlyReportActualMonth),
  }];
}

/**
 * Obtain the status color of resume, showing the respectively colors:
 * Sell Volume R$0-R$15_000.00 = Ok-Green
 * Sell Volume R$15_000.00-R$18_000.00 = Alert-Yellow
 * Sell Volume R$18_000.00-R$20_000.00 or more = Warning-Red
 * @param monthlyReport: actual month data that will be used for resume
 */
export function getStatusResume(monthlyReport : MonthlyTaxReport) : EResumeStatus {
  const { remainExemptionVolumeBrazil } = monthlyReport;
  if (remainExemptionVolumeBrazil > 5_000) {
    return EResumeStatus.GOOD;
  }
  if (remainExemptionVolumeBrazil > 2_000) {
    return EResumeStatus.WARNING;
  }
  return EResumeStatus.BAD;
}

/**
 * Discover the status color of calendar monthly card using the tax volume.
 *  Month in progress - White
 *  No Darf - Green
 *  Darf Paid - Blue
 *  Darf Pending - Yellow
 *  Darf Late - Red
 * @param tax: volume of tax.
 * @param month: number of month, to discover if is early or in future of actual month
 * @param darfPaid: boolean to discover if darf is paid or not
 */
export function getTaxStatus(tax: number, month : number, darfPaid : boolean) : EProfitCardStatus {
  const actualMonth = new Date().getMonth();

  if (month >= actualMonth) {
    return EProfitCardStatus.NOT_FINISHED;
  }

  if (tax === 0 || tax < 10) {
    return EProfitCardStatus.NO_DARF;
  }

  if (darfPaid) {
    return EProfitCardStatus.PAID;
  }

  if (month === actualMonth - 1) {
    return EProfitCardStatus.PENDING;
  }

  return EProfitCardStatus.LATE;
}
