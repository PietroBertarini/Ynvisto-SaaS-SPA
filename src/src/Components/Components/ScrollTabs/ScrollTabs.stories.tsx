import React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Meta, Story } from '@storybook/react/types-6-0';
import ScrollTabs, { ScrollTabsProps } from './ScrollTabs.component';
import { mockTabs } from '../../../Mocks/ScrollTabs.mock';

export default {
  title: 'Components/ScrollTabs',
  component: ScrollTabs,
} as Meta;

const Template: Story<ScrollTabsProps> = (args) => <ScrollTabs {...args} />;

export const Default = Template.bind({});
Default.args = {
  initialTabValue: 0,
  tabsLabel: mockTabs,
};
