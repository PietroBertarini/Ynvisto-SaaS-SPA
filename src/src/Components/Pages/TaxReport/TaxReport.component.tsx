import React, { useEffect } from 'react';
import { Typography } from '@material-ui/core';
import { connect, ConnectedProps, useDispatch } from 'react-redux';
import '../../../Theme/sweetalert2/sweetalert2.scss';
import { Redirect } from 'react-router-dom';
import { useStyles } from './TaxReport.styles';
import { State } from '../../../Redux/root-reducer';
import ProfitCalendar from '../../Components/ProfitCalendar/ProfitCalendar.component';
import ScrollTabs from '../../Components/ScrollTabs/ScrollTabs.component';
import Resume from '../../Components/Resume/Resume.component';
import ProfitActualMonth from '../../Components/ProfitActualMonth/ProfitActualMonth.component';
import ProfitCalendarLegend, {
  EColor,
  ILegendData,
} from '../../Components/ProfitCalendarLegend/ProfitCalendarLegend.component';
import { fetchTaxReportsRequest } from '../../../Redux/Services/TaxReports/TaxReports.actions';
import Spinner from '../../Components/Spinner/spinner.component';
import { allTabs, buildProfitCardsProps, getResumeInfo } from './TaxReport.utils';

export const calendarHelper : ILegendData[] = [
  {
    label: 'DARF Paga',
    color: EColor.BLUE,
  },
  {
    label: 'Não houve DARF',
    color: EColor.GREEN,
  },
  {
    label: 'DARF Pendente',
    color: EColor.YELLOW,
  },
  {
    label: 'DARF Atrasada',
    color: EColor.RED,
  },
  {
    label: 'Mês não finalizado',
    color: EColor.WHITE,
  },
];

/**
 * Get the stores states that TaxComponents uses and observe.
 * @param state: the stored stated of the Ynvisto.
 */
const mapStateToProps = (state: State) => ({
  taxReport: state.taxReports.taxReport,
  isFetching: state.taxReports.isFetchingReport,
  spreadsheetAdm: state.authorization.spreadsheetAdmin,
  error: state.taxReports.error,
});

const connector = connect(mapStateToProps);
type PropsFromRedux = ConnectedProps<typeof connector>
type Props = PropsFromRedux

function TaxReport(props: Props) {
  const dispatch = useDispatch();
  const classes = useStyles();
  const {
    taxReport, isFetching, error, spreadsheetAdm,
  } = props;

  useEffect(() => {
    if (taxReport === undefined) {
      dispatch(fetchTaxReportsRequest(spreadsheetAdm));
    }
    return () => {
      // Clean previously data for fetch again yhe most updated tax report data.
      // dispatch(taxReportCleanData());
    };
  }, [dispatch, taxReport, spreadsheetAdm]);
  if (isFetching) {
    return <Spinner />;
  }

  // TODO: Find a way to make this check standard across all API Calls, just like the error building and checking is
  if (error?.httpStatusCode === 401) {
    return <Redirect to="/Logout/" />;
  }

  const actualMonth : number = new Date().getMonth();
  const profitCardsData = buildProfitCardsProps(taxReport?.monthlyReports);
  const resumeInfo = getResumeInfo(taxReport?.monthlyReports);
  return (
    <div className={classes.root}>
      <Typography className={classes.header}>
        Apuração de Lucros
      </Typography>
      <ScrollTabs initialTabValue={0} tabsLabel={allTabs} />
      <Resume allResumeInfo={resumeInfo} />
      <Typography className={classes.subTitle}>
        Mês Atual:
      </Typography>
      <ProfitActualMonth month={profitCardsData[actualMonth]} />
      <Typography className={classes.subTitle}>
        Todos os meses:
      </Typography>
      <ProfitCalendarLegend legendData={calendarHelper} />
      <ProfitCalendar monthArray={profitCardsData} />
    </div>
  );
}
export default connector(TaxReport);
