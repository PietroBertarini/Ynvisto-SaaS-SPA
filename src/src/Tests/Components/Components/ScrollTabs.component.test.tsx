import React from 'react';
import { shallow } from 'enzyme';
import { allTabs } from '../../../Components/Pages/TaxReport/TaxReport.utils';
import ScrollTabs from '../../../Components/Components/ScrollTabs/ScrollTabs.component';

test('Renders ScrollTabs component and match to previously snapshot', () => {
  const component = shallow(<ScrollTabs initialTabValue={0} tabsLabel={allTabs} />);
  expect(component).toMatchSnapshot();
});
