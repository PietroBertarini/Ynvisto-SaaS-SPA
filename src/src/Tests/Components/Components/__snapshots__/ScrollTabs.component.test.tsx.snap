// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ScrollTabs component and match to previously snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    color="transparent"
    elevation={0}
    position="static"
  >
    <WithStyles(Component)
      aria-label="scrollable auto tabs example"
      className="makeStyles-styledTabs-2"
      indicatorColor="secondary"
      onChange={[Function]}
      scrollButtons="auto"
      textColor="secondary"
      value={0}
      variant="scrollable"
    >
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-0"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-0"
        key="Geral"
        label="Geral"
        to="Geral"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-1"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-1"
        key="Janeiro"
        label="Janeiro"
        to="Janeiro"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-2"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-2"
        key="Fevereiro"
        label="Fevereiro"
        to="Fevereiro"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-3"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-3"
        key="Março"
        label="Março"
        to="Março"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-4"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-4"
        key="Abril"
        label="Abril"
        to="Abril"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-5"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-5"
        key="Maio"
        label="Maio"
        to="Maio"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-6"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-6"
        key="Junho"
        label="Junho"
        to="Junho"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-7"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-7"
        key="Julho"
        label="Julho"
        to="Julho"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-8"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-8"
        key="Agosto"
        label="Agosto"
        to="Agosto"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-9"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-9"
        key="Setembro"
        label="Setembro"
        to="Setembro"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-10"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-10"
        key="Outubro"
        label="Outubro"
        to="Outubro"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-11"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-11"
        key="Novembro"
        label="Novembro"
        to="Novembro"
      />
      <WithStyles(Component)
        aria-controls="scrollable-auto-tabpanel-12"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        id="scrollable-auto-tab-12"
        key="Dezembro"
        label="Dezembro"
        to="Dezembro"
      />
    </WithStyles(Component)>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
